source/slang/core.meta.slang

```diff
@@ -652,23 +653,36 @@ for (int tt = 0; tt < kBaseTextureTypeCount; ++tt)
                 if (isArray) sb << "Array";
                 sb << "<T> t, ";
                 sb << "SamplerState s);\n";
+
+                sb << "__target_intrinsic(glsl, texture)\n";
+                sb << "T Sample(float" << kBaseTextureTypes[tt].coordCount + isArray << " location);\n";
+
+                sb << "__target_intrinsic(glsl, texture)\n";
+                sb << "T Sample(float" << kBaseTextureTypes[tt].coordCount + isArray << " location);\n";
+
                 sb << "};\n";
 
-                sb << "__specialized_for_target(glsl)\n";
-                sb << "T texture<T>(Sampler";
-                sb << kBaseTextureAccessLevels[accessLevel].name;
-                sb << name;
-                if (isMultisample) sb << "MS";
-                if (isArray) sb << "Array";
-                sb << "<T> t, float" << kBaseTextureTypes[tt].coordCount + isArray << " location);\n";
             }
         }
     }
 }
+#endif
+
+static const struct TextureTypePrefixInfo
+{
+    char const* name;
+    bool        combined;
+} kTexturePrefixes[] =
+{
+    { "Texture", false },
+    { "Sampler", true },
+};
 
+for(auto& prefixInfo : kTexturePrefixes)
```

Comment:
It seems perhaps a little irksome it's not indented - but in the context of the complexity/interation over variety of cases I guess it's fair enough. It's like you want 'iterate over the combinations of some things'.

Reply:
I'm not *proud* of doing that, but I also wanted to avoid the massive diff due to re-indenting some of our most complicated stdlib generation logic.

It actually seems like we might benefit from factoring the logic that generates the texture type definitions into one or more subroutines, or a "context" type with overloadable policy methods. There's just *so much* detail there and it is basically impossible to get anything right on the first try.